recipes/GigaSpeech/gigaspeech_prepare.py

```diff
@@ -0,0 +1,703 @@
+    punctuation: bool,
+    stopwords: list,
+) -> list:
+    """
+    Process the audio line and return the utterances for the given split.
+
+    Parameters
+    ----------
+    audio : dict
+        The audio line to be processed.
+    data_folder : str
+        The path to the GigaSpeech dataset.
+    split : str
+        The split to be used for filtering the data.
+    convert_opus_to_wav : bool
+        If True, the opus files will be converted to wav files.
+    punctuation : bool
+        Keeping punctuation or not. Default is no.
+    stopwords: list
+        List of stopwords to remove from the text of the labels.
+
+    Returns
+    -------
+    list
+        The list of utterances for the given split.
+    """
+    if ("{" + split + "}") in audio["subsets"]:
+
+        audio_path = os.path.join(data_folder, audio["path"])
+        assert os.path.isfile(audio_path), f"File not found: {audio_path}"
+
+        if convert_opus_to_wav and audio_path.endswith(".opus"):
+            audio_path = convert_opus2wav(audio_path)
+
+        # 2. iterate over the utterances
+        utterances = []
+        for segment in audio["segments"]:
+            text = preprocess_text(segment["text_tn"], punctuation, stopwords)
+            if text:
+                begin_time = float(segment["begin_time"])
+                end_time = float(segment["end_time"])
+                duration = end_time - begin_time
+                utterance = GigaSpeechRow(
+                    utt_id=segment["sid"],
+                    audio_id=audio["aid"],
+                    audio_path=str(audio_path),
+                    speaker=audio["speaker"],
+                    begin_time=begin_time,
+                    end_time=end_time,
+                    duration=duration,
+                    text=text,
+                )
+                utterances.append(utterance)
+        return utterances
+
+
+def create_csv(
+    csv_file: str,
+    info: json,
+    data_folder: str,
+    split: str,
+    convert_opus_to_wav: bool,
+    punctuation: bool = False,
+    filler: bool = False,
+) -> None:
+    """
+    Create a CSV file based on the info in the GigaSpeech JSON file and filter the data based on the split.
+
+    Parameters
+    ----------
+    csv_file : str
+        The path to the CSV file to be created.
+    info : dict
+        The GigaSpeech JSON file content.
+    data_folder : str
+        The path to the GigaSpeech dataset.
+    split : str
+        The split to be used for filtering the data.
+    convert_opus_to_wav : bool
+        If True, the opus files will be converted to wav files.
+    punctuation : bool
+        Keeping punctuation or not. Default is no.
+    filler : bool
+        Keeping filler words or not (hum, er). Default is no.
+
+    Returns
+    -------
+    None
+    """
+    total_duration = 0.0
+    nb_samples = 0
+
+    to_remove = GARBAGE_UTTERANCE_TAGS
+    if not punctuation:
+        to_remove += PUNCTUATION_TAGS
```

Comment:
Not correct because this is a dict. Shouldn't this be removed because of how this is handled in `preprocess_text`, anyway?